src/main/java/seedu/edrecord/logic/commands/EditCommand.java

```diff
@@ -230,12 +230,17 @@ public boolean equals(Object other) {
 
             // state check
             EditPersonDescriptor e = (EditPersonDescriptor) other;
+            Optional<Module> thisModule = getModule();
+            Optional<Module> otherModule = e.getModule();
+            boolean isModuleEqual = (thisModule.equals(otherModule)) // default equality of Optional<T> objects
+                    || (thisModule.isPresent() && otherModule.isPresent() // or both modules are present
+                    && thisModule.get().isSameModule(otherModule.get())); // and have the same identity
 
             return getName().equals(e.getName())
                     && getPhone().equals(e.getPhone())
                     && getEmail().equals(e.getEmail())
                     && getInfo().equals(e.getInfo())
-                    && getModule().equals(e.getModule())
+                    && isModuleEqual
```

Comment:
Why need to define this new boolean to check module equality? Just curious cuz this is checking equality for EditCommands which should be ok with just the normal .equals() no?

Reply:
This one is because `getModule()` returns an `Optional<Module>`, and I want to check for equality using `isSameModule` instead of `Module#equals()`, so need to have a separate boolean to check.